.changelog/371.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/ec_deployment: Fix a bug affecting partial version upgrades, where some resources upgraded but others did not, leaving the `version` argument set to the configured value as if the upgrade was successful. The upgrade could not be retried and it was unable to detect drift.
```

Comment:
```suggestion
resource/ec_deployment: Fixed a bug that affects partial version upgrades. During an upgrade only a subset of resources would upgrade successfully, but the `version` argument value updated as if all resources were upgraded. Attempts to retry the upgrade would fail since the version difference was not detected.
```
I think Arianna is offline at the moment so I'd suggest the above for the bug text.

Reply:
Thanks!